mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1855,6 +1855,140 @@ func TestClientSideEncryptionProse(t *testing.T) {
 		})
 
 	})
+
+	mt.RunOpts("16. Rewrap", runOpts, func(mt *mtest.T) {
+		mt.Run("Case 1: Rewrap with separate ClientEncryption", func(mt *mtest.T) {
+			dataKeyMap := map[string]bson.M{
+				"aws": {
+					"region": "us-east-1",
+					"key":    "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0",
+				},
+				"azure": {
+					"keyVaultEndpoint": "key-vault-csfle.vault.azure.net",
+					"keyName":          "key-name-csfle",
+				},
+				"gcp": {
+					"projectId": "devprod-drivers",
+					"location":  "global",
+					"keyRing":   "key-ring-csfle",
+					"keyName":   "key-name-csfle",
+				},
+				"kmip": {},
+			}
+
+			tlsConfig := make(map[string]*tls.Config)
+			if tlsCAFileKMIP != "" && tlsClientCertificateKeyFileKMIP != "" {
+				tlsOpts := map[string]interface{}{
+					"tlsCertificateKeyFile": tlsClientCertificateKeyFileKMIP,
+					"tlsCAFile":             tlsCAFileKMIP,
+				}
+				kmipConfig, err := options.BuildTLSConfig(tlsOpts)
+				assert.Nil(mt, err, "BuildTLSConfig error: %v", err)
+				tlsConfig["kmip"] = kmipConfig
+			}
+
+			kmsProviders := []string{"local", "aws", "gcp", "azure", "kmip"}
+			for _, srcProvider := range kmsProviders {
+				for _, dstProvider := range kmsProviders {
+					mt.Run(fmt.Sprintf("%s to %s", srcProvider, dstProvider), func(mt *mtest.T) {
+						var err error
+						// Drop the collection ``keyvault.datakeys``.
+						{
```

Comment:
Hmm is the point of these empty blocks just to emphasize what the steps of this test are? Or are you using them for some scope logic that I'm not understanding 🧑‍🔧 . If it's just for emphasizing the steps, I'm not sure it's necessary.

Reply:
Yes, it is to emphasize what code accomplishes the step. It is not necessary.